### PR TITLE
Pinning distroless images to specific SHA. Fixes #20997

### LINKED ISSUE #20997: Pin distroless base images to a particular distroless SHA
Per discussion up to https://github.com/istio/istio/pull/20904#issuecomment-583904080, we should probably pin the distroless images used in the docker builds to a particular distroless SHA.

### PR DESCRIPTION
**Pull request by stewartbutler**
Distroless does not produce sane tags, so to get a stable image we need to use a SHA. This PR just takes a snapshot of the current `latest` tag of each image.

Fixes #20997

### CODE REVIEW

File: mixer/test/listbackend/cmd/Dockerfile
Comment (howardjohn) [inline mixer/test/listbackend/cmd/Dockerfile]: nit: remove these

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
There are many other usages as well, did you specifically not change them?
```
security/docker/Dockerfile.citadel
security/docker/Dockerfile.node-agent
mixer/docker/Dockerfile.test_policybackend
mixer/docker/Dockerfile.mixer
mixer/docker/Dockerfile.mixer_codegen
sidecar-injector/docker/Dockerfile.sidecar_injector
pilot/docker/Dockerfile.proxyv2
pilot/docker/Dockerfile.pilot
galley/docker/Dockerfile.galley
operator/docker/Dockerfile.operator
samples/tcp-echo/src/Dockerfile
mixer/test/listbackend/cmd/Dockerfile
mixer/test/prometheus/cmd/Dockerfile
```

**Comment by stewartbutler** [comment]
My IDE's search did not find them. Grep did, though -- good catch

**Review by howardjohn — APPROVED** [review]

**Review by myidpt — APPROVED** [review]

**Comment by sdake** [comment]
@stewartbutler thank you for the contribution! It's good to see community members using distroless. Can you rebase this PR so we can merge?

Cheers
-steve

**Comment by sdake** [comment]
@stewartbutler or @ericvn some followup work after this PR merges. Place the distroless tag in the core Makefile, and pass this as a build argument.

Cheers
-steve

**Review by elfinhe — APPROVED** [review]

**Comment by stewartbutler** [comment]
@sdake Rebased, can I get a signoff for docs maintainers?

@mandarjog Can you take a look and signoff for policy/telemetry?

**Review by ericvn — APPROVED** [review]

**Review by mandarjog — APPROVED** [review]
/lgtm

**Pull request merged**
